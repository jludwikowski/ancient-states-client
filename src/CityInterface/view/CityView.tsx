import React from "react"
import { BuildingBrowser } from "../components/BuildingBrowser";
import Building from "../../Models/Building";
import { defaultBuildingList } from "../../BackEndMock/buildings"

export class CityView extends React.Component {

    buildingList: Building[] = defaultBuildingList;

    componentDidMount(): void {

    };

    onBuild(id:number) {

    };

    render() {
        return(
            <>
                <BuildingBrowser buildingList={this.buildingList} onBuild={this.onBuild} />
            </>
    )}
}
