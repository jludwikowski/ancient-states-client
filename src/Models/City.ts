import Building from './Building';
import {Entity} from './Entity';

export default interface City extends Entity {
    buildings: Building[],
    id: number,
    owner: number,
    numberOfMen: number,
    numberOfWomen: number,
    numberOfSlaves: number,
    constructing: string | null,
    constructionTimeLeft: number | null,
    position: string,
    barracksLevel: number,
    forgeLevel: number,
    fieldsLevel: number,
    wallLevel: number,
}
