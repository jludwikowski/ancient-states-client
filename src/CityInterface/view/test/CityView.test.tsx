import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import City from '../../../Models/City';
import { CityView } from '../CityView';



describe('CityView', () => {

    const city:City = {
        name: 'Persepolis',
        description: 'Greatest City in the World',
        numberOfSlaves: 100000,
        numberOfWomen: 30000,
        numberOfMen: 30000,
        id: 1,
        constructing: null,
        constructionTimeLeft: null,
        barracksLevel: 10,
        forgeLevel: 10,
        fieldsLevel: 30,
        wallLevel: 30,
        position: '0,0',
        owner: 1,
        buildings: [
            {
                baseBuildingTime: 120,
                baseCost: 100,
                description: 'You recruit standard units here',
                id: 1,
                imageUrl: 'images/icons/icons8-military-base-100.png',
                level: 10,
                name: 'barracks'
            },
            {
                baseBuildingTime: 150,
                baseCost: 250,
                description: 'This building let you upgrade weapons',
                id: 2,
                imageUrl: 'images/icons/icons8-military-base-100.png',
                level: 10,
                name: 'forge'
            },
            {
                baseBuildingTime: 150,
                baseCost: 60,
                description: 'This building produce food',
                id: 3,
                imageUrl: 'images/icons/icons8-military-base-100.png',
                level: 30,
                name: 'fields'
            },
        ],
};

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<CityView {...city} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<CityView {...city} />);
        expect(component).toMatchSnapshot();
    });
});
