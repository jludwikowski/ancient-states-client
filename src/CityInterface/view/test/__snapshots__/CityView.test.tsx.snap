// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityView Should match snapshot 1`] = `
<Fragment>
  <BuildingBrowser
    buildingList={
      Array [
        Object {
          "baseBuildingTime": 120,
          "baseCost": 100,
          "description": "You recruit standard units here",
          "id": 1,
          "imageUrl": "images/icons/icons8-military-base-100.png",
          "level": 10,
          "name": "barracks",
        },
        Object {
          "baseBuildingTime": 150,
          "baseCost": 250,
          "description": "This building let you upgrade weapons",
          "id": 2,
          "imageUrl": "images/icons/icons8-military-base-100.png",
          "level": 10,
          "name": "forge",
        },
        Object {
          "baseBuildingTime": 150,
          "baseCost": 60,
          "description": "This building produce food",
          "id": 3,
          "imageUrl": "images/icons/icons8-military-base-100.png",
          "level": 30,
          "name": "fields",
        },
      ]
    }
    onBuild={[Function]}
  />
</Fragment>
`;
